test/box-luatest/gh_7202_options_in_atomic_test.lua

```diff
@@ -0,0 +1,202 @@
+
+        -- Concurrent atomics with function and table.
+        local fibers = {}
+        local results = {}
+
+        fiber.create(function()
+            table.insert(fibers, fiber.self())
+            fiber.self():set_joinable(true)
+            local succ, err = pcall(box.atomic, f, 'atomic1')
+            results[1] = {succ, tostring(err)}
+        end)
+
+        fiber.create(function()
+            table.insert(fibers, fiber.self())
+            fiber.self():set_joinable(true)
+            local succ, err = pcall(box.atomic, t, 'atomic2')
+            results[2] = {succ, tostring(err)}
+        end)
+
+        lt.assert_equals(#fibers, 2)
+        for _,f in pairs(fibers) do
+            f:join()
+        end
+        lt.assert_equals(results,
+            {{true, 'nil'},
+             {false, 'Transaction has been aborted by conflict'}})
+        lt.assert_equals(box.space.kv:select{}, {{'last_id', 3}})
+        lt.assert_equals(box.space.test:select{},
+            {{1, 'test1', 'function'},
+             {2, 'test2', 'table'},
+             {3, 'atomic1', 'function'}})
+    end)
+end
+
+-- Atomic with options
+g.test_atomic_with_options = function()
+    g.server:exec(function()
+        local lt = require('luatest')
+        local fiber = require('fiber')
+
+        local function get_next_id()
+            local t = box.space.kv:get{'last_id'}
+            local id = t[2] + 1
+            box.space.kv:replace{'last_id', id}
+            return id
+        end
+
+        local function f(data)
+            local id = get_next_id()
+            box.space.test:replace{id, data, 'function'}
+        end
+        local mt = {
+            __call = function(t, data)
+                assert(t ~= nil)
+                local id = get_next_id()
+                box.space.test:replace{id, data, 'table'}
+            end
+        }
+        local t = setmetatable({}, mt)
+
+        local fibers = {}
+        local results = {}
+        local opts = {txn_isolation = box.txn_isolation_level.READ_COMMITTED}
+
+        fiber.create(function()
+            table.insert(fibers, fiber.self())
+            fiber.self():set_joinable(true)
+            local succ, err = pcall(box.atomic, opts, f, 'atomic1')
+            results[1] = {succ, tostring(err)}
+        end)
+
+        fiber.create(function()
+            table.insert(fibers, fiber.self())
+            fiber.self():set_joinable(true)
+            local succ, err = pcall(box.atomic, opts, t, 'atomic2')
+            results[2] = {succ, tostring(err)}
+        end)
+
+        lt.assert_equals(#fibers, 2)
+        for _,f in pairs(fibers) do
+            f:join()
+        end
+        lt.assert_equals(results,
+            {{true, 'nil'},
+             {true, 'nil'}})
+        lt.assert_equals(box.space.kv:select{}, {{'last_id', 2}})
+        lt.assert_equals(box.space.test:select{},
+            {{1, 'atomic1', 'function'},
+             {2, 'atomic2', 'table'}})
+    end)
+end
```

Comment:
I thought we had a function to get the current transaction's isolation level, but unfortunately we do not have one. If we did — I would rather test that the isolation level is correct.

TBH, it's kind of hard for to understand what's going on these tests: as I understand, you are basically testing that the isolation levels work correctly — but we already have [gh-6390](https://github.com/tarantool/tarantool/blob/master/test/engine-luatest/gh_6930_mvcc_isolation_levels_test.lua) test for this.